js/errors.go

```diff
@@ -0,0 +1,109 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2019 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package js
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/dop251/goja"
+
+	"go.k6.io/k6/errext"
+	"go.k6.io/k6/errext/exitcodes"
+	"go.k6.io/k6/lib/consts"
+)
+
+// scriptException is used for errors that resulted from a script exception and
+// contain a stack trace that lead to them.
+type scriptException struct {
+	inner *goja.Exception
+}
+
+var _ errext.Exception = &scriptException{}
```

Comment:
You should probably check for the `HasHint` and `HasErrorCode` while you are at it.

Reply:
although to be honest I am not certain if I wouldn't have preferred there to be `newScriptExceiption()` which calls 

`WithHint(WithExitCode(scriptException{inner: ex}...))` 

On the other hand my biggest problem is that errext.ScriptException implementations can have different exitcodes ... which again is the whole problem with the exit codes being codified by whatever returns them instead of what will exit with them ...

Reply:
Decided not to wrap this in `WithHint(WithExitCode())`, it didn't really improve things in terms of readability or clarity